### ISSUE TITLE
Kernel module error, openrazer not working

### ISSUE THREAD

**Issue by RayZ3R0** [openrazer/razer-drivers#2177]
### Support

- [x] Have you tried rebooting? -> Yes
- [x] Have you looked at the [Troubleshooting](https://github.com/openrazer/openrazer/wiki/Troubleshooting) page? -> Yes

I installed polychromatic and openrazer, they had no issues when installing, but after troubleshooting and using the commands listed, I'm getting these errors.
![image](https://github.com/openrazer/openrazer/assets/69317757/678370ad-aaa1-4fd5-a7df-04a91b6f4439)
![image](https://github.com/openrazer/openrazer/assets/69317757/99565656-4156-4b5c-a36a-93eaae381d75)
![image](https://github.com/openrazer/openrazer/assets/69317757/04908e10-6fc1-4d18-a257-ded0ee475487)

**Comment by z3ntu** [comment]
What happens when you run `sudo dnf install kernel-devel` ?

**Comment by Sid127** [comment]
just ran into this on Fedora 40 as well... the kernel modules do not get built automatically on installing `openrazer-meta`
I had to manually build and install them via dkms. Hopefully this can be fixed soon

**Comment by z3ntu** [comment]
We do have the dkms commands in the Fedora package so they should get called https://github.com/openrazer/OBS-packaging/blob/master/openrazer.spec#L154-L171

Did you maybe see an error or something during first Installation?

**Comment by Sid127** [comment]
Nope, did not see any errors. Looking at the dkms command, it seems to be hitting an edge case where it builds only for the running kernel, and not every kernel on the disk.

My installation looked like
- Fresh install of F40 KDE
- dnf update -y
- installation of nvidia modules and openrazer
- reboot (to a new kernel version)

EDIT: I now realize my issue is different from OP's 😅 Apologies for that.


-------- Original Message --------

**Comment by z3ntu** [comment]
Right, the man page also says that...

> If the kernel option is not specified it assumes the currently running kernel.

Unfortunately the `dkms build` / `dkms install` command doesn't support `--all` which would be what we want then.. Maybe we need to manually iterate over the installed kernels and provide them with `-k`?

See also https://github.com/dell/dkms/issues/400 / https://github.com/dell/dkms/issues/394

**Comment by z3ntu** [comment]
@Sid127 Are you interested in working on this? We probably just need a few lines of bash added around the lines I linked in the .spec file above that gets a list of installed kernels to build against, and then we iterate in `dkms install` over them. I don't think we need to adjust `dkms remove` part at all.

**Comment by Sid127** [comment]
sure, I'll take a look at it tomorrow and make a Pull Request accordingly

-------- Original Message --------

**Comment by z3ntu** [comment]
Thanks, appreciate it!

**Comment by Sid127** [comment]
oh shit I forgot about this

**Comment by z3ntu** [comment]
That's another perk of having a stale bot 😉  (though it should've triggered earlier but due to a configuration error the bot never made it to this older issue in its run)

I'll mark it as a bug & packaging-related now, seems you know/knew what was wrong.
